cdap-hbase-compat-1.1/src/main/java/co/cask/cdap/data2/replication/hbase11/LastWriteTimeObserver.java

```diff
@@ -67,8 +71,8 @@ public void postWALWrite(ObserverContext<? extends WALCoprocessorEnvironment> ct
               logKey.getTablename().toString(),
               logKey.getWriteTime(),
               logKey.getEncodedRegionName().toString());
-    if (tableUpdater != null) {
-      tableUpdater.updateTime(new String(logKey.getEncodedRegionName(), "UTF-8"), logKey.getWriteTime());
+    if (hBase11TableUpdater != null) {
+      hBase11TableUpdater.updateTime(new String(logKey.getEncodedRegionName(), "UTF-8"), logKey.getWriteTime());
```

Comment:
No! You cannot decode something that isn't a string into a string, encode that string as bytes, and expect to get the same thing back. To see this, try doing:

```
    byte[] val = Bytes.toBytes(System.currentTimeMillis());
    String valStr = Bytes.toString(val);
    byte[] valStrBytes = Bytes.toBytes(valStr);
```

This encoded key has a start key (not a string) and a timestamp (not a string).